src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -301,12 +380,77 @@ public void setBaitSetName(final String name) {
+            if (rec.getMateAlignmentStart() < rec.getAlignmentStart()) return rec; // right-most, so ignore.  Do this first since getAlignmentEnd is expensive
+            final int numBasesToClip = rec.getMateAlignmentStart() - rec.getAlignmentEnd() + 1;
+            if (0 < numBasesToClip) return rec; // left-most but not overlapping
+
+            try {
+                // watch out for when the second read overlaps all of the first read
+                if (rec.getMateAlignmentStart() <= rec.getAlignmentStart()) { // make it unmapped
+                    if (noSideEffects) {
+                        final SAMRecord recClone = ((SAMRecord)rec.clone());
+                        recClone.setReadUnmappedFlag(true);
+                        return recClone;
+                    }
+                    else rec.setReadUnmappedFlag(true);
+                    return rec;
+                }
+
+                // clip it, clip it good
+                final Cigar cigar = CigarUtil.addSoftClippedBasesToEndsOfCigar(
+                        rec.getCigar(),
+                        false, // ignore, since we want 3' end genomic order
+                        numBasesToClip, // want 3' end genomic order
+                        0
+                );
+                if (noSideEffects) {
+                    final SAMRecord recClone = ((SAMRecord)rec.clone());
+                    recClone.setCigar(cigar);
+                    return recClone;
+                }
+                else {
+                    rec.setCigar(cigar);
+                    return rec;
+                }
+            } catch (final CloneNotSupportedException e) {
+                throw new PicardException(e.getMessage(), e);
+            }
+        }
+
         /** Adds information about an individual SAMRecord to the statistics. */
-        public void acceptRecord(final SAMRecord rec) {
-            // Just plain avoid records that are marked as not-primary
-            if (rec.isSecondaryOrSupplementary()) return;
+        public void acceptRecord(final SAMRecord record) {
+            // We should not "Just plain avoid records that are marked as not-primary" since:
+            // 1. We should disregard secondary alignments in for both record and base counting.
+            // 2. We should disregard supplementary alignments for ONLY record counting
+            // This means that we need to consider supplemental alignments for base counting but not record counting.
+
+            // NB: this could modify the record.  See noSideEffects
+            final SAMRecord rec;
+            if (clipOverlappingReads) rec = clipOverlappingBases(record);
+            else rec = record;
+
+            // apply any filters
+            for (final SamRecordFilter filter : this.filters) {
+                if (filter.filterOut(rec)) return;
+            }
 
-            this.metrics.TOTAL_READS += 1;
+            if (!rec.getSupplementaryAlignmentFlag()) this.metrics.TOTAL_READS += 1;
```

Comment:
This is really a great point.  It would really annoy me if, for example, ```TOTAL_READS``` and ```PF_READS``` did not match the values computed found ```AlignmentSummaryMetrics```.  However, the latter ignores all secondary and supplementary records, with is inappropriate for counting how many bases aligned, so perhaps I need to update that?